index.js

```diff
@@ -612,4 +617,18 @@ HtmlWebpackPlugin.prototype.getAssetFiles = function (assets) {
   return files;
 };
 
+/**
+ * Helper to promisify compilation.applyPluginsAsyncWaterfall that returns
+ * a function that helps to merge given plugin arguments with processed ones
+ */
+HtmlWebpackPlugin.prototype.applyPluginsAsyncWaterfall = function (compilation) {
+  var promisedApplyPluginsAsyncWaterfall = Promise.promisify(compilation.applyPluginsAsyncWaterfall, {context: compilation});
+  return function (eventName, pluginArgs) {
+    return promisedApplyPluginsAsyncWaterfall(eventName, pluginArgs)
+      .then(function (result) {
+        return _.extend(pluginArgs, result);
```

Comment:
Maybe I am just confused but the error occurs only if there are two plugins and the first one doesn't return a result right?

So if only one plugin is used which returns no value this should show the deprecated warning but still everything would work.